### ISSUE TITLE
Support for <img/> elements using `object-fit`

### ISSUE THREAD

**Issue by nathanielrindlaub** [ZeeCoder/use-resize-observer#106]
I am trying to use use-resize-observer to watch for changes to the rendered dimensions of an `<img/>` element that uses `object-fit: contain`, but it appears as though the returned width & height dimensions are that of that image's parent element, not the rendered image itself. Is this intended behavior? Some sample code is below:

```javascript
import React, { useRef, useState } from "react";
import ReactDOM from "react-dom";
import useResizeObserver from "use-resize-observer";
import "./styles.css";
import image from '../public/deer-mouse.jpg';

const App = () => {
  const imgRef = useRef(null);
  const { width, height } = useResizeObserver({ ref: imgRef });

  return (
    <>
      <div className="measurement">
        imgRef size is:{" "}
        <b>
          {width}x{height}
        </b>
      </div>
      <div>
        <div className="box">
          <img ref={imgRef} style={{ width: '100%', height: '100%', objectFit: 'contain'}} src={image} />
        </div>
      </div>
    </>
  );
};

const rootElement = document.getElementById("root");
ReactDOM.render(<App />, rootElement);
```

**Comment by ZeeCoder** [comment]
It may be, this hook is a very thin wrapper around ResizeObserver.
Have you tried using a native ResizeObserve instance to see if you're getting anything different from it compared to the hook?

Could you create a codesandbox where the difference in behaviour is showcased?

It can also depend on the browser, [see this svg issue](https://github.com/ZeeCoder/use-resize-observer/issues/85#issuecomment-1575750260).

**Comment by nathanielrindlaub** [comment]
Hey @ZeeCoder - I just put together a [codesandbox example](https://codesandbox.io/p/sandbox/use-resize-observer-observe-img-with-object-fit-9jwxnp?file=%2Fsrc%2Findex.js%3A31%2C6&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clpit307400072e66679sr87x%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clpit307300032e668ur24dif%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clpit307300042e66urr3d7o5%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clpit307400062e66utnl0gwf%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clpit307300032e668ur24dif%2522%253A%257B%2522id%2522%253A%2522clpit307300032e668ur24dif%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clpx04loo00022e66btpwpwru%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A31%252C%2522startColumn%2522%253A6%252C%2522endLineNumber%2522%253A31%252C%2522endColumn%2522%253A6%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clpx04loo00022e66btpwpwru%2522%257D%252C%2522clpit307400062e66utnl0gwf%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clpit307300052e66wo93uet4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clpit307400062e66utnl0gwf%2522%252C%2522activeTabId%2522%253A%2522clpit307300052e66wo93uet4%2522%257D%252C%2522clpit307300042e66urr3d7o5%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clpit307300042e66urr3d7o5%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) and it appears as though ResizeObserver exhibits the same behavior.

I double checked against a couple different browsers and they all did the same thing.

**Comment by ZeeCoder** [comment]
Thanks for the codesandbox!
I think I see now where the confusion is.
While object fit changes the size of the image visibly, the element's size itself doesn't change, as you can see from the outline.

see this screenshot where I added a red border to the img element, and added a 20px margin around it so that the separation between it and the parent is clear:
![image](https://github.com/ZeeCoder/use-resize-observer/assets/3679506/a15e2edc-637f-4425-83af-58d0cd75be21)


Basically the resize observer measures the box size correctly.

I'm no css wizard unfortunately, but there might be a different way to do this where the box size follows the img size as well.

You could also just measure the size of the parent with uRO, and cacluate the size of the contained image yourself with JS, since you're already measuring things.

**Comment by nathanielrindlaub** [comment]
Ah gotcha - thanks for looking into it! Yeah, for now I am just calculating the size of the contained image myself and it's working fine.

**Comment by ZeeCoder** [comment]
Glad I could help :muscle:
